README.md

```diff
@@ -259,6 +259,33 @@ h Name=refs/heads/master Repo=terraform-aws-eks
+git-xargs performs all work up to issuing the API call to open a pull request in parallel - according to the concurrency limits the user might optionally specify by way of the `--max-concurrent-repos` flag.
+
+Meanwhile, git-xargs flows all open pull request jobs through a separate time-buffered channel.
+
+This means that git-xargs performs all the work upfront that it can as quickly as possible, and then moves onto serially process the pull request jobs that have resulted from the upfront work of cloning reposotories, making file changes, checking the git worktree, etc.
+
+**Automatic pull request retries when rate limited**
+By default, git-xargs will re-attempt opening a pull request that failed due to rate limiting. The `--max-pull-request-retries` flag allows you to specify how many times you'd like a given pull request to be re-attempted in case of failure due to rate limiting. By default, the value is `3`, meaning that if you do not pass this flag, `git-xargs` will retry all rate-limit-blocked pull requests 3 times.
+
+**Automatic backoff when rate limiting is detected**
+When git-xargs detects that it has been rate limited by GitHub, it begins requeuing failed pull requests for retry, but with an additional, larger buffer of time in between the next attemp. This larger buffer of time is intended to allow GitHub rate limit status to return to normal. The `--seconds-to-wait-when-ratelimited` flag specifies the number of seconds in addition to wait when git-xargs detects it has been rate limited. Note that this extra buffer of time is in addition to the value specified by the `--seconds-between-prs` flag.
+
+When it is provided by GitHub, git-xargs will instead use the values of any `Retry-After` header, or the delta of seconds between the current time and the rate limit error's reset time. When these two values are not available, git-xargs falls back to the user-specified value of `--seconds-to-wait-when-ratelimited`, if the flag was passed, or the default value, which is 60 seconds.
+
+**Specifying the pause between pull requests**
+The GitHub's API [integration guidelines](https://docs.github.com/en/rest/guides/best-practices-for-integrators#dealing-with-secondary-rate-limits) specify that clients should pause at least 1 second in between consecutive requests against expensive endpoints, such as the one for opening a new pull request. As a result, git-xargs defaults to pausing 1 second in between each pull request that is opened. The flag `--seconds-between-prs` allows you to modify this value, so, for example, if you were to pass `--seconds-between-prs 30`
+
+**Guidelines and observations from testing**
+In local testing, GitHub's secondary rate limits tend to kick in quickly once you're running git-xargs on more than 30 repositories concurrently. In such scenarios, passing a `--seconds-between-prs` value of 12 seconds tends to provide sufficient buffer between open pull request API calls to avoid being rate limited.
+
```

Comment:
The whole readme is 💯, well-written, easy to understand, so thank you!

Thank you for including these two sections to explain when to use the bare minimum 1 second vs 12 or 30 seconds.

Reply:
Thank you, Rho!